File: server/src/main/java/bit/project/server/entity/Valuation.java
package bit.project.server.entity;


import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.List;


@Data
@Entity
@NoArgsConstructor
@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
public class Valuation {
    @Id
    @GeneratedValue(strategy=GenerationType.IDENTITY)
    private Integer id;

    private String code;

    private String chassisnumber;

    private String enginenumber;

    private LocalDate yearofmanufacture;

    private LocalDate dateofregistration;

    private String color;

    private String originalcolor;

    @Lob
    private String description;

    private String enginecapacity;

    @ManyToOne
    private Fueltype fueltype;

    @ManyToOne
    private Vehiclemodel vehiclemodel;

    @ManyToOne
    private Province province;

    private String registrationno;

    private LocalDateTime tocreation;

    @ManyToOne
    private Vehiclecondition vehiclecondition;

    @ManyToOne
    private Country country;


    @ManyToOne
    private Vehicletype vehicletype;

    @ManyToOne
    private Brand brand;

    private Integer odometerreading;

    private BigDecimal marketvalue;

    private BigDecimal forcedsalesvalue;

    private LocalDate valuationdate;

    @ManyToOne
    private Valuationorganization valuationorganization;

    @ManyToOne
    private Vehicle vehicle;

    @ManyToOne
    private Valuationstatus valuationstatus;

    @JsonIgnore
    @OneToMany(mappedBy = "valuation")
    private List<Offerrequest> valuationList;


    @ManyToOne
    @JsonIgnoreProperties({"creator","tocreation","roleList"})
    private User creator;


    public Valuation(Integer id) {
        this.id = id;
    }

    public Valuation(Integer id, String code, String registrationno, String chassisnumber, String enginenumber){
        this.code = code;
        this.registrationno = registrationno;
        this.chassisnumber = chassisnumber;
        this.enginenumber = enginenumber;
        this.id = id;

    }

}

File: client/src/app/entities/valuation.ts
import {User} from './user';
import {DataPage} from '../shared/data-page';
import {Province} from './province';
import {Vehiclemodel} from './vehiclemodel';
import {Vehiclecondition} from './vehiclecondition';
import {Fueltype} from './fueltype';
import {Vehicletype} from './vehicletype';
import {Country} from './country';
import {Brand} from './brand';
import {Valuationorganization} from './valuationorganization';
import {Valuationstatus} from './valuationstatus';
import {Vehicle} from './vehicle';

export class Valuation {
  id: number;
  code: string;
  chassisnumber: string;
  enginenumber: string;
  yearofmanufacture: string;
  dateofregistration: string;
  color: string;
  originalcolor: string;
  description: string;
  enginecapacity: string;
  fueltype: Fueltype;
  vehiclemodel: Vehiclemodel;
  province: Province;
  registrationno: string;
  tocreation: string;
  vehiclecondition: Vehiclecondition;
  creator: User;
  country:Country;
  vehicletype: Vehicletype;
  brand: Brand;
  odometerreading: number;
  marketvalue: number;
  forcedsalesvalue: number;
  valuationdate: string;
  valuationorganization: Valuationorganization;
  valuationstatus: Valuationstatus;
  vehicle: Vehicle;



  constructor(id: number = null) {
    this.id = id;
  }
}
export class ValuationDataPage extends DataPage{
    content: Valuation[];
}
